src/TableOperations.jl

```diff
@@ -309,9 +311,33 @@ function joinpartitions(x)
             N = length(schema[].names)
             foreach(i -> push!(joined, ChainedVector([Tables.getcolumn(cols, i)])), 1:N)
         else
-            foreach(i -> append!(joined[i], Tables.getcolumn(cols, i)), 1:N)
+            foreach(1:N) do i
```

Comment:
I'll admit that this is a bit uglier and may trigger extra allocations, but that should be relatively infrequently.